cinder/volume/drivers/vmware/fcd.py

```diff
@@ -492,7 +492,11 @@ def copy_image_to_volume(self, context, volume, image_service, image_id):
         profile_id = self._get_storage_profile_id(volume)
         if profile_id:
             self.volumeops.update_fcd_policy(fcd_loc, profile_id)
-
+        try:
+            LOG.warning("Extending volume %s to %s", volume.id, volume.size)
+            self.volumeops.extend_fcd(fcd_loc, volume['size'] * units.Ki)
```

Comment:
If we create a volume from snapshot or from fcd, we call `self._extend_if_needed()` and I'm wondering if we shouldn't do that here, too.

Additionally, we usually use `volume.size` - are you certain that `volume['size']` will be fine, too? Looking at the code, we have 10 times `volume.size` and 14 times `volume['size']` ... weirdly inconsistent. But probably works.

Reply:
the volume object attributes work either way with .size or ['size']. 

Also, extend_if_needed tests between 2 sizes.  For this case where we are creating a volume from image, the image.size is the correct size, so there is no check against cur_size and new_size, because volume.size is the right size it should be.

Reply:
posted a follow up PR for this